### ISSUE TITLE
NullPointerException in ResourceServiceManager when no automatic package replication is set up

### ISSUE THREAD

**Issue by csaboka** [Adobe-Consulting-Services/acs-aem-commons#1064]
When the ConfigurationUpdateListener class gets activated, it causes a NPE if there are no automatic package replication pages set up:

```
02.08.2017 18:34:32.359 *ERROR* [FelixStartLevel] com.adobe.acs.acs-aem-commons-bundle [com.adobe.acs.commons.replication.packages.automatic.impl.ConfigurationUpdateListener(2991)] The activate method has thrown an exception (java.lang.NullPointerException)
java.lang.NullPointerException: null
    at com.adobe.acs.commons.util.ResourceServiceManager.refreshCache(ResourceServiceManager.java:155)
    at com.adobe.acs.commons.util.ResourceServiceManager.activate(ResourceServiceManager.java:70)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
(rest of stack trace omitted for brevity)

```
Line 155 tries to access the serviceReferences array for logging, but that array can be null if there are no replication pages present and therefore no service references exist.

The issue is more serious than it first looks. The Sling OsgiObservationBridge binds to every component that uses the legacy event-based observation to warn about them, so it tries to bind to ConfigurationUpdateListener as well, and the failure to bind causes the whole bridge to fail. This means _no_ legacy observation events are delivered at all, which is how I noticed the issue in the first place.

The fix should be simple enough: check the array for null, and if it is null, replace it with a zero-length array instead.

**Comment by davidjgonzalez** [comment]
We should fix for imminent 3.10.0 release

**Comment by csaboka** [comment]
Thank you very much!
